backend/loi-service/src/main/java/org/egov/works/service/WorkflowService.java

```diff
@@ -0,0 +1,250 @@
+package org.egov.works.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import digit.models.coremodels.*;
+import org.apache.commons.lang3.StringUtils;
+import org.egov.common.contract.request.RequestInfo;
+import org.egov.common.contract.request.User;
+import org.egov.tracer.model.CustomException;
+import org.egov.works.config.LOIConfiguration;
+import org.egov.works.repository.ServiceRequestRepository;
+import org.egov.works.web.models.LetterOfIndent;
+import org.egov.works.web.models.LetterOfIndentRequest;
+import org.egov.works.web.models.LetterOfIndentRequestWorkflow;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+public class WorkflowService {
+
+    @Autowired
+    private LOIConfiguration serviceConfiguration;
+
+    @Autowired
+    private ServiceRequestRepository repository;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+
+    /*
+     * Should return the applicable BusinessService for the given request
+     *
+     */
+    public BusinessService getBusinessService(LetterOfIndentRequest loiRequest) {
+        String tenantId = loiRequest.getLetterOfIndent().getTenantId();
+        StringBuilder url = getSearchURLWithParams(tenantId, serviceConfiguration.getLoiWFBusinessService());
+        RequestInfo requestInfo = loiRequest.getRequestInfo();
+        Object result = repository.fetchResult(url, requestInfo);
```

Comment:
Do we need this api call? 
I had asked @pintu-eGov to remove this. 
Please check the estimate-service code. This api call might have been skipped and a better error message provided in the transition api call. 

Even in the current code, I don't think you are calling this function.